### PR TITLE
add ngram tokenizer docs

### PR DESCRIPTION
**Pull request by AntonEliatra**
### Description
add ngram tokenizer docs

### Issues Resolved
Part of https://github.com/opensearch-project/documentation-website/issues/1483 addressed in this PR.

### Version
all

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by vagimeli** [comment]
@udabhas @varun-lodaya Please provide tech review approval to move this PR forward in the documentation process. Please review this week or provide a peer who can review it. Thank you.

File: _analyzers/tokenizers/ngram.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+}
+```
+
+## Parameters
+
+The `ngram` tokenizer can be configured with the following parameters.
+
+Parameter | Required/Optional | Data type | Description
+:--- | :--- | :--- | :--- 
+`min_gram` | Optional | Integer | Minimum length of n-grams. Default is `1`.
```
Comment (natebower) [inline _analyzers/tokenizers/ngram.md:76]:
```suggestion
`min_gram` | Optional | Integer | The minimum length of the n-grams. Default is `1`.
```

File: _analyzers/tokenizers/ngram.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+`max_gram` | Optional | Integer | Maximum length of n-grams. Default is `2`.
```
Comment (natebower) [inline _analyzers/tokenizers/ngram.md:77]:
```suggestion
`max_gram` | Optional | Integer | The maximum length of the n-grams. Default is `2`.
```

File: _analyzers/tokenizers/ngram.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+`token_chars` | Optional | List of strings | Character classes to be included in tokenization. Valid values are:<br>- `letter`<br>- `digit`<br>- `whitespace`<br>- `punctuation`<br>- `symbol`<br>- `custom` (You must also specify the `custom_token_chars` parameter)<br>Default is empty list (`[]`), which retains all the characters.
```
Comment (natebower) [inline _analyzers/tokenizers/ngram.md:78]:
```suggestion
`token_chars` | Optional | List of strings | The character classes to be included in tokenization. Valid values are:<br>- `letter`<br>- `digit`<br>- `whitespace`<br>- `punctuation`<br>- `symbol`<br>- `custom` (You must also specify the `custom_token_chars` parameter)<br>Default is an empty list (`[]`), which retains all the characters.
```

File: _analyzers/tokenizers/ngram.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+`custom_token_chars` | Optional | String | Custom characters to be included as part of the tokens.
```
Comment (natebower) [inline _analyzers/tokenizers/ngram.md:79]:
```suggestion
`custom_token_chars` | Optional | String | Custom characters to be included in the tokens.
```

File: _analyzers/tokenizers/ngram.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+### Maximum difference between `min_gram` and `max_gram`
+
+The maximum difference between `min_gram` and `max_gram` is configured using the index-level `index.max_ngram_diff` setting and defaults to `1`.
+
+The following example creates index with a custom `index.max_ngram_diff` setting:
```
Comment (natebower) [inline _analyzers/tokenizers/ngram.md:85]:
```suggestion
The following example request creates an index with a custom `index.max_ngram_diff` setting: 
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws @AntonEliatra Editorial review. Thanks!

**Review by kolchfa-aws — APPROVED** [review]

**Pull request merged**